src/frontend/src/test-e2e/selenium.test.ts

```diff
@@ -55,453 +41,368 @@ const DEMO_APP_URL = "http://localhost:8080/";
-      expect(alertText).toBe(
+      await browser2.waitUntil(async () => !!(await browser2.getAlertText()), {
+        timeout: 1_000,
+        timeoutMsg: "expected alert to be displayed after 1s",
+      });
+      expect(await browser2.getAlertText()).toBe(
         "This will remove your current device and you will be logged out"
       );
+      await browser2.dismissAlert();
+      await browser2.deleteSession();
     });
 
     // About page
-    await driver.get("about:blank");
-    await driver.get(II_URL + "#about");
-    await wait_for_fonts(driver);
-    const aboutView = new AboutView(driver);
+    await browser.url("about:blank");
+    await browser.url(II_URL + "#about");
+    await waitForFonts(browser);
+    const aboutView = new AboutView(browser);
     await aboutView.waitForDisplay();
-    await screenshot("14-about", driver);
+    await screenshot("14-about", browser);
 
     // Test device removal
-    await driver.get(II_URL);
+    await browser.url(II_URL);
     await welcomeBackView.waitForDisplay();
     const userNumber3 = await welcomeBackView.getIdentityAnchor();
     expect(userNumber3).toBe(userNumber);
     await welcomeBackView.login();
     await singleDeviceWarningView.waitForDisplay();
     await singleDeviceWarningView.continue();
     await recoveryMethodSelectorView.waitForDisplay();
-    await recoveryMethodSelectorView.skip();
+    await recoveryMethodSelectorView.skipRecovery();
     await mainView.waitForDeviceDisplay(DEVICE_NAME2);
-    const buttonElem2 = await driver.findElement(
-      By.xpath(`//div[string()='${DEVICE_NAME2}']/following-sibling::button`)
-    );
     await mainView.removeDevice(DEVICE_NAME2);
     // No dialog here!
-
-    await driver.wait(until.stalenessOf(buttonElem2));
+    await browser.waitUntil(async () => !(await browser.getAlertText()), {
+      timeout: 3_000,
+      timeoutMsg: "expected alert to NOT be displayed after 3s",
+    });
+    await browser.waitUntil(
+      async () => {
+        const device2 = await browser.$(`//div[string()='${DEVICE_NAME2}']`);
+        return !(await device2.isDisplayed());
+      },
+      {
+        timeout: 5_000,
+        timeoutMsg: "expected device2 to be gone after 5s",
+      }
+    );
     await mainView.waitForDeviceDisplay(DEVICE_NAME1);
     await mainView.fixup();
-    await screenshot("15-after-removal", driver);
+    await screenshot("15-after-removal", browser);
 
     await mainView.removeDevice(DEVICE_NAME1);
-    const alert1 = driver.switchTo().alert();
-    const alertText1 = await alert1.getText();
-    await alert1.accept();
+    const alertText1 = await browser.getAlertText();
     expect(alertText1).toBe(
       "This will remove your current device and you will be logged out"
     );
-    await driver.wait(until.alertIsPresent());
-    const alert2 = driver.switchTo().alert();
-    const alertText2 = await alert2.getText();
-    await alert2.accept();
+    await browser.acceptAlert();
```

Comment:
Shouldn't this be done after we checked the alter text?

Reply:
I kept the logic as it was before. There are two alerts in succession and it seems to sufficient to wait for the second one